src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -135,8 +157,11 @@ public override void OnStartActivity(Activity activity, object payload)
                     return;
                 }
 
-                ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
-                ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Server);
+                if (!IsNet7OrGreater)
```

Comment:
why we have used pre-processor directives (NET7_0_OR_GREATER) in some places and the static variable in other place?